sqlite/migrate.go

```diff
@@ -183,11 +184,84 @@ func GetLastMigrationVersion(db *sql.DB) (version uint, migrationTableExists boo
 
 	var lastMigration uint64 = 0
 	if migrationTableExists {
-		row = db.QueryRow("SELECT version FROM status_go_schema_migrations")
+		row = db.QueryRow(fmt.Sprintf("SELECT version FROM %s", migrationTableName))
 		err = row.Scan(&lastMigration)
 		if err != nil && err != sql.ErrNoRows {
 			return 0, true, err
 		}
 	}
 	return uint(lastMigration), migrationTableExists, nil
 }
+
+// Used to migrate from one migration table to another, ensuring the table reflects the correct migration state.
+// Ensures migrationTableName exists and records the latest version from assetNames, capped at maxVersion.
+// Intended for migration table transitions and version synchronization.
+func UpdateMigrationTableVersion(db *sql.DB, migrationTableName string, assetNames []string, maxVersion uint) error {
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			rollbackErr := tx.Rollback()
+			if rollbackErr != nil {
+				err = fmt.Errorf("failed to rollback transaction: %w; original error: %v", rollbackErr, err)
+			}
+		}
+	}()
+
+	row := tx.QueryRow("SELECT exists(SELECT name FROM sqlite_master WHERE type='table' AND name=?)", migrationTableName)
+	exists := false
+	err = row.Scan(&exists)
+	if err != nil && err != sql.ErrNoRows {
+		return err
+	}
+
+	if !exists {
+		createTable := fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (version uint64, dirty bool);`, migrationTableName)
+		_, err = tx.Exec(createTable)
+		if err != nil {
+			return err
+		}
+		createIndex := fmt.Sprintf(`CREATE UNIQUE INDEX IF NOT EXISTS version_unique ON %s (version);`, migrationTableName)
+		_, err = tx.Exec(createIndex)
+		if err != nil {
+			return err
+		}
+	}
+
+	version := getMaxMigrationVersion(assetNames, maxVersion)
+	if version > 0 {
+		// #nosec G201 -- migrationTableName is a trusted constant, not user input
+		deleteQuery := fmt.Sprintf("DELETE FROM %s", migrationTableName)
+		_, err = tx.Exec(deleteQuery)
+		if err != nil {
+			return err
+		}
+
+		insertVersion := fmt.Sprintf(`INSERT INTO %s (version, dirty)`, migrationTableName) + `VALUES (?, ?)`
+		_, err = tx.Exec(insertVersion, version, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = tx.Commit()
+
+	return err
+}
+
+func getMaxMigrationVersion(assetNames []string, max uint) uint {
+	floor := uint(0)
+	for _, name := range assetNames {
+		m, err := source.DefaultParse(name)
+		if err != nil {
+			continue // ignore files that we can't parse
+		}
+		if m.Version <= max && (floor == 0 || m.Version > floor) {
```

Comment:
I believe we don't have any negative migration versions, so when `floor == 0`, then `m.Version > floor` 
```suggestion
		if m.Version <= max && m.Version > floor {
```